### PR TITLE
Improve bootstrap logging for help with iframe debugging

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

This should help with issues like https://github.com/lensapp/lens/issues/4018

### CODE REVIEW

File: src/common/base-store.ts [hunk 1]
```diff
@@ -43,6 +43,8 @@ export abstract class BaseStore<T> extends Singleton {
   protected storeConfig?: Config<T>;
   protected syncDisposers: Disposer[] = [];
 
+  readonly displayName: string = this.constructor.name;
```
Comment (ixrock) [inline src/common/base-store.ts:46]: Remember info for class name would be useless/muggled when app is built with webpack's `--production=true` && `optimization.minimized = true`

Comment (Nokel81) [reply src/common/base-store.ts:46]: Yes I know, which is why for all of our built-in `T extends BaseStore` I override this. I would have wanted to make it `abstract` but that would have been a breaking change.

File: src/common/base-store.ts [hunk 2]
```diff
@@ -66,13 +72,13 @@ export abstract class BaseStore<T> extends Singleton {
     const res: any = this.fromStore(this.storeConfig.store);
 
     if (res instanceof Promise || (typeof res === "object" && res && typeof res.then === "function")) {
-      console.error(`${this.constructor.name} extends BaseStore<T>'s fromStore method returns a Promise or promise-like object. This is an error and must be fixed.`);
+      console.error(`${this.displayName} extends BaseStore<T>'s fromStore method returns a Promise or promise-like object. This is an error and must be fixed.`);
```
Comment (jim-docker) [inline src/common/base-store.ts:75]: errors must go through console?

File: src/renderer/root-frame.tsx [hunk 3]
```diff
@@ -55,11 +58,10 @@ export class LensApp extends React.Component {
 
     registerIpcListeners();
 
-    window.onbeforeunload = () => {
-      logger.info("[App]: Unload app");
-
+    window.addEventListener("beforeunload", () => {
+      logger.info(`${RootFrame.logPrefix} Unload app`);
```
Comment (jim-docker) [inline src/renderer/root-frame.tsx:62]:
```suggestion
      logger.info(`${RootFrame.logPrefix} Unload root frame`);
```
?

**Review by jim-docker — APPROVED** [review]

File: src/common/base-store.ts (hunk 2 above)
Comment (Nokel81) [reply src/common/base-store.ts:75]: Do you mean instead of `logger`? That will be resolved by #4317

File: src/renderer/root-frame.tsx (hunk 3 above)
Comment (Nokel81) [reply src/renderer/root-frame.tsx:62]: I don't think that is necessary given that `logPrefix` is already `"[ROOT-FRAME]:"`

**Pull request merged**